src/Microsoft.JSInterop/DotNetObjectRef.cs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+
+namespace Microsoft.JSInterop
+{
+    /// <summary>
+    /// Wraps a JS interop argument, indicating that the value should not be serialized as JSON
+    /// but instead should be passed as a reference.
+    ///
+    /// To avoid leaking memory, the reference must later be disposed by JS code or by .NET code.
+    /// </summary>
+    public class DotNetObjectRef : IDisposable
```

Comment:
Should this have a finalizer?

Is this class just infrastructure? I'm not sure I understand yet how this is used

Reply:
Yeah I supposed the finalizer wouldn't help in this case because its rooted by the runtime